File: MDEV-1004-S23-ICE8b/Server/Models/movie.ts
// -------------------------------------------------------------
// - Robert Bettinelli - MDEV1004 - S2023 - Assignment#1
// - 090003683@student.georgianc.on.ca
// -------------------------------------------------------------
// (Models) movie.ts - As Provided in Class Instruction
// Personally entered and followed as pert of in class learning.
// -------------------------------------------------------------
// 06/10/2023 - RBettinelli - Header and Documentation Added
// -------------------------------------------------------------

import { Schema, model } from "mongoose";

// Movie interface for Collecting data in specific types. 
interface IMovie {
    movieID: number,
    title: string,
    studio: string,
    genres: string[],
    directors: string[],
    writers: string[],
    actors: string[],
    year: number,
    length: string,
    shortDescription: string,
    mpaRating: string,
    criticsRating: number,
    posterLink: string
}

// Mongo DB schema Setup
const movieSchema = new Schema<IMovie>({
    movieID: {
      type: Number,
      required: true,
    },
    title: {
      type: String,
      required: true,
    },
    studio: {
      type: String,
      required: true,
    },
    genres: {
      type: [String],
      required: true,
    },
    directors: {
      type: [String],
      required: true,
    },
    writers: {
      type: [String],
      required: true,
    },
    actors: {
      type: [String],
      required: true,
    },
    year: {
      type: Number,
      required: true, 
    },
    length: {
      type: String,
      required: true, 
    },
    shortDescription: {
      type: String,
      required: true, 
    },
    mpaRating: {
      type: String,
      required: true, 
    },
    criticsRating: {
      type: Number,
      required: true, 
    },
    posterLink: {
      type: String,
      required: true, 
    },
  });

// Build Model Object as Interface utilizing Schema. 
let Movie = model<IMovie>('movies', movieSchema);

export default Movie;

File: MDEV-1004-S23-A4/MDEV1004S23A4/app/src/main/java/com/example/mdev1004s23A4/Movie.kt
package com.example.mdev1004s23A4

// -------------------------------------------------------------
// - Robert Bettinelli - MDEV1004 - S2023
// - 090003683@student.georgianc.on.ca
// -------------------------------------------------------------
// 08/23/2023 - RBettinelli - Header and Documentation Added
// -------------------------------------------------------------
// Contains Data Structures for JSON Calls.
// -------------------------------------------------------------
import java.io.Serializable

data class Movie(
    val _id: String,
    val movieID: Int,
    val title: String,
    val studio: String,
    val genres: List<String>,
    val directors: List<String>,
    val writers: List<String>,
    val actors: List<String>,
    val year: Int,
    val length: String,
    val shortDescription: String,
    val mpaRating: String,
    val criticsRating: Double,
    val posterLink: String
) : Serializable

data class MovieResponse(
    val success: Boolean,
    val msg: String,
    val data: MutableList<Movie>
)

data class MovieResponseWrapper(
    val success: Boolean,
    val msg: String,
    val data: Movie
)

data class MovieDeleteWrapper(
    val success: Boolean,
    val msg: String,
    val data: String
)

data class PaymentWrapper(
    val success: Boolean,
    val msg: String,
    val clientSecret: String,
    val customer: String,
    val publishableKey: String
)